pkg/controller/netflow.go

```diff
@@ -0,0 +1,169 @@
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/client-go/tools/cache"
+
+	"github.com/noironetworks/aci-containers/pkg/apicapi"
+)
+
+const (
+	netflowCRDName = "netflowpolicies.aci.netflow"
+)
+
+type ContNetflowType struct {
+	DstAddr           string `json:"dstAddr"`
+	DstPort           string `json:"dstPort"`
+	Ver               string `json:"type"`
+	ActiveFlowTimeOut int    `json:"activeFlowTimeOut"`
+	IdleFlowTimeOut   int    `json:"idleFlowTimeOut"`
+}
+
+type NetflowPolicy struct {
+	FlowSamplingPolicy ContNetflowType
+}
+
+func NetflowPolicyLogger(log *logrus.Logger, netflow *netflowpolicy.NetflowPolicy) *logrus.Entry {
+	return log.WithFields(logrus.Fields{
+		"name": netflow.ObjectMeta.Name,
+		"spec": netflow.Spec,
+	})
+}
+
+func netflowInit(cont *AciController, stopCh <-chan struct{}) {
+	cont.log.Debug("Initializing netflow client")
+	restconfig := cont.env.RESTConfig()
+	netflowClient, err := netflowclientset.NewForConfig(restconfig)
+	if err != nil {
+		cont.log.Errorf("Failed to intialize netflow client")
+		return
+	}
+	cont.initNetflowInformerFromClient(netflowClient)
+	cont.log.Infof("Netflow cache synced")
+	cont.netflowInformer.Run(stopCh)
+}
+
+func (cont *AciController) initNetflowInformerFromClient(
+	netflowClient *netflowclientset.Clientset) {
+	cont.initNetflowInformerBase(
+		cache.NewListWatchFromClient(
+			netflowClient.AciV1alpha().RESTClient(), "netflowpolicies",
+			metav1.NamespaceAll, fields.Everything()))
+}
+
+func (cont *AciController) initNetflowInformerBase(listWatch *cache.ListWatch) {
+	cont.netflowIndexer, cont.netflowInformer = cache.NewIndexerInformer(
+		listWatch,
+		&netflowpolicy.NetflowPolicy{}, 0,
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				cont.netflowPolicyUpdated(obj)
+			},
+			UpdateFunc: func(_, obj interface{}) {
+				cont.netflowPolicyUpdated(obj)
+			},
+			DeleteFunc: func(obj interface{}) {
+				cont.netflowPolicyDelete(obj)
+			},
+		},
+		cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc},
+	)
+	cont.log.Debug("Initializing Netflow Policy Informers")
+
+}
+
+func (cont *AciController) netflowPolicyUpdated(obj interface{}) {
+	netflowPolicy := obj.(*netflowpolicy.NetflowPolicy)
+	key, err := cache.MetaNamespaceKeyFunc(netflowPolicy)
+	if err != nil {
+		NetflowPolicyLogger(cont.log, netflowPolicy).
+			Error("Could not create key:" + err.Error())
+		return
+	}
+	cont.queueNetflowUpdateByKey(key)
+
+}
+
+func (cont *AciController) queueNetflowUpdateByKey(key string) {
+	cont.netflowQueue.Add(key)
+}
+
+func (cont *AciController) queueNetflowUpdate(netflowpolicy *netflowpolicy.NetflowPolicy) {
+	key, err := cache.MetaNamespaceKeyFunc(netflowpolicy)
+	if err != nil {
+		NetflowPolicyLogger(cont.log, netflowpolicy).
+			Error("Could not create key:" + err.Error())
+		return
+	}
+	cont.netflowQueue.Add(key)
+}
+
+func (cont *AciController) netflowPolicyDelete(netflowobj interface{}) {
+	nf := netflowobj.(*netflowpolicy.NetflowPolicy)
```

Comment:
Could you please follow this example to handle delete in services.go func serviceDeleted

Reply:
To clarify, should I make changes in services.go by following this example or make changes in netflow.go by following serviceDeleted example?

Reply:
Addressed it